### PR TITLE
Add support for Flight Availabilities Search API

### LINKED ISSUE #101: Add support for Flight Availabilities Search API
https://developers.amadeus.com/self-service/category/air/api-doc/flight-availabilities-search/api-reference

### PR DESCRIPTION
**Pull request by xianqiliu**
Fixes #101 
@tsolakoua @gimperial 
This PR is ready for review, please go ahead and thank you!

### CODE REVIEW

File: src/main/java/com/amadeus/resources/FlightAvailabilitySearch.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  private @Getter String id;
+  private @Getter String originalDestinationId;
+  private @Getter String source;
+  private @Getter Boolean instantTicketingRequired;
+  private @Getter Boolean paymentCardRequired;
+  private @Getter String duration;
+  private @Getter SearchSegment[] segments;
+
+  @ToString
+  public class SearchSegment {
```
Comment (gimperial) [inline src/main/java/com/amadeus/resources/FlightAvailabilitySearch.java:26]: I think you have missed some elements from the reply object.
I can't see the closedStatus, co2Emissions, operating, duration and stops

**Comment by gimperial** [comment]
@xianqiliu Can you please include the API in the readMe as well as some test scenarios? Thank you

**Comment by gimperial** [comment]
@tsolakoua Is there anything to do related to the X-HTTP-Method-Override?

File: src/main/java/com/amadeus/resources/FlightAvailabilitySearch.java (hunk 1 above)
Comment (xianqiliu) [reply src/main/java/com/amadeus/resources/FlightAvailabilitySearch.java]: Thank you for noticing that! I have added the missed elements by following the model in the [doc](https://developers.amadeus.com/self-service/category/air/api-doc/flight-availabilities-search/api-reference).

**Comment by xianqiliu** [comment]
> @xianqiliu Can you please include the API in the readMe as well as some test scenarios? Thank you

Yes, already added to ` List of Supported Endpoints` in the readme.

**Pull request closed without merging**